snuba_sdk/conditions.py

```diff
@@ -59,3 +59,35 @@ def validate(self) -> None:
             raise InvalidExpression(
                 f"invalid condition: RHS of a condition must be a Column, CurriedFunction, Function or Scalar not {type(self.rhs)}"
             )
+
+
+class BooleanOp(Enum):
+    AND = "AND"
+    OR = "OR"
+
+
+@dataclass(frozen=True)
+class BooleanCondition(Expression):
+    op: BooleanOp
+    conditions: Sequence[Union["BooleanCondition", Condition]]
```

Comment:
I like the idea of simplifying the syntax for conditions. I wonder if this needs to be a first class expression type (which has its own class and thus it is visible to the visitor).
Have you consider the alternative of building the DSL on top of the existing expressions like `and(condition, condition, etc.)` which would be a function that produces a `FunctionCall("AND", etc.)`. 
Which keeps the first class expressions to deal with simpler (simpler visitors).
If yes, what was the trade off? And what was the reasoning to make BooleanCondition a first class expression?

To be clear, the alternative I am bringing up would still be part of the sdk repo, just not a dedicated expression type.